internal/ioextensions/network_error_folder.go

```diff
@@ -0,0 +1,71 @@
+package ioextensions
+
+import (
+	"io"
+	"sync"
+
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+func NewUnexpectedEOFLimitReader(readCloser io.ReadCloser, maxRead int64) io.ReadCloser {
+	return &UnexpectedEOFLimitReader{
+		source: readCloser,
+		left:   maxRead,
+	}
+}
+
+type UnexpectedEOFLimitReader struct {
+	source io.ReadCloser
+	left   int64
+}
+
+func (u *UnexpectedEOFLimitReader) Read(p []byte) (n int, err error) {
+	if u.left <= 0 {
+		return 0, io.ErrUnexpectedEOF
+	}
+	unexpectedEOF := false
+	if int64(len(p)) > u.left {
+		unexpectedEOF = true
+		p = p[:u.left]
+	}
+	n, err = u.source.Read(p)
+	u.left -= int64(n)
+	if unexpectedEOF && err == nil {
+		err = io.ErrUnexpectedEOF
+	}
+	return
+}
+
+func (u *UnexpectedEOFLimitReader) Close() error {
+	return u.source.Close()
+}
+
+func NewNetworkErrorFolder(sourceFolder storage.Folder, networkErrorAfterByteSize int) storage.Folder {
```

Comment:
Is it used for tests only ? 
It's better to put it in the corresponding `_test.go` to avoid using in non-test code in the future.